### PR TITLE
vm/editor

### PR DESCRIPTION
**Pull request by parvdesai-tenable**
Contains method related to vm/editors endpoint

### CODE REVIEW

File: tenable/io/v3/vm/__init__.py [hunk 1]
```diff
@@ -82,6 +84,14 @@ def credentials(self):
         '''
         return CredentialsAPI(self._api)
 
+    @property
+    def editor(self):
+        '''
+        The interface object for the
+        :doc:`Tenable.IO v3 API <editor>`
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/__init__.py]: Tenable.io v3 Editor APIs <editor>

File: tenable/io/v3/vm/editor/api.py [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+
+        Args:
+            etype (str):
+                The type of template to retrieve.  Must be either ``scan`` or
+                ``policy``.
+            object_id (int):
+                The unique identifier of the object.
+            file_id (int):
+                The unique identifier of the file to export.
+            fobj (FileObject):
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/editor/api.py]: Please update it to
fobj (FileObject, optional)

File: tenable/io/v3/vm/editor/api.py [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+        Args:
+            etype (str): The type of object to request.
+            scan_id (int): The unique identifier for the scan.
+
+        Returns:
+            :obj:`dict`:
+                The constructed scan configuration resource.
+
+        Examples:
+            >>> policy = tio.editor.details('scan', 1)
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/editor/api.py]: This should be tio.v3.vm.editor.details...

File: tenable/io/v3/vm/editor/api.py [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+                for plugin in plugs:
+                    plugins[plugin['id']] = plugin['status']
+                resp[family] = {
+                    'mixedDefault': 'enabled',
+                    'status': 'mixed',
+                    'individual': plugins,
+                }
+        return resp
+
+    def plugin_details(self,
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/editor/api.py]: Let's keep the name same as v2: **plugin_description**.

**Review by dhvanilshah-tenable — APPROVED** [review]

File: tenable/io/v3/vm/editor/schema.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class EditorTemplateSchema(Schema):
+    '''
+    Schema for editor template
+    '''
+    etype = fields.Str(
+        required=True,
+        validate=validate.OneOf(['scan', 'policy', 'remediation'])
+    )
```
Comment (milanaghera-tenable) [inline tenable/io/v3/vm/editor/schema.py:31]: can't we merge these 3 schemas into a single schema and then using validate using the validate_schema decorator

File: tests/io/v3/vm/editor/test_editor_api.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+'''
+from pathlib import Path
+
+import requests
+import responses
+
+from tenable.io.v3.base.iterators.explore_iterator import (CSVChunkIterator,
+                                                           SearchIterator)
+
+from .objects import (CONFIGURATION_DETAILS, CONFIGURATION_ID, ETYPE,
```
Comment (milanaghera-tenable) [inline tests/io/v3/vm/editor/test_editor_api.py]: here you can simply use `from .objects import *` and also use the absolute path for import

**Review by milanaghera-tenable — APPROVED** [review]

File: tenable/io/v3/vm/editor/schema.py (hunk 5 above)
Comment (parvdesai-tenable) [reply tenable/io/v3/vm/editor/schema.py:31]: that would be same as not using marshmallow.

File: tests/io/v3/vm/editor/test_editor_api.py (hunk 6 above)
Comment (parvdesai-tenable) [reply tests/io/v3/vm/editor/test_editor_api.py]: absolute path fixed. star import will cause flake8 failures

**Pull request merged**